manager/controllers/app/blueprint_controller.go

```diff
@@ -141,7 +155,53 @@ func (r *BlueprintReconciler) applyChartResource(log logr.Logger, chartSpec app.
 	nbytes, _ := yaml.Marshal(args)
 	log.Info(fmt.Sprintf("--- Values.yaml ---\n\n%s\n\n", nbytes))
 
+	var registrySuccessfulLogin string
+
+	if chartSpec.ChartPullSecret != "" {
+		// obtain ChartPullSecret
+		pullSecret := corev1.Secret{}
+		pullSecrets := []corev1.Secret{}
+
+		if err := r.Get(ctx, types.NamespacedName{Namespace: utils.GetSystemNamespace(), Name: chartSpec.ChartPullSecret},
+			&pullSecret); err == nil {
+			// if this is not a dockerconfigjson, ignore
+			if pullSecret.Type == "kubernetes.io/dockerconfigjson" {
+				pullSecrets = append(pullSecrets, pullSecret)
+			}
+		}
+
+		// create a keyring of all dockerconfigjson secrets, to be used for lookup
```

Comment:
At this point pullSecrets can be empty. You could try to pull chart anyway, but the credentialprovider functionality is not required in this case, right?

Reply:
right. I will try to simplify